launcher/FileSystem.cpp

```diff
@@ -369,10 +369,19 @@ QString getDesktopDir()
 bool createShortcut(QString destination, QString target, QStringList args, QString name, QString icon)
 {
 #if defined(Q_OS_MACOS)
+    QString invalidMacOSChars = ":";
```

Comment:
This check ends up repeating a lot of code.
I think it would  be better to place only a `QString invalidOSChars =  "what needs to be here";`
inside the appropriate pre-processor if blocks and write the check loop only once.

Reply:
created a separate function in https://github.com/PrismLauncher/PrismLauncher/pull/826/commits/fd2a8efde8f59f3db1bdb28832ad3e5f7b2e70f0